plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -257,3 +261,25 @@ def _get_filter_time(self, custom_config: Dict[str, int], normal_running_cutoff:
         self.logger.info(f"The following filter time will be used: {filter_time}")
 
         return filter_time
+
+    def check_rate_limit(self, state: Dict):
+        rate_limited = state.get(self.RATE_LIMIT_DATETIME)
+        now = time()
+        if rate_limited:
+            rate_limit_string = Utils.convert_epoch_to_readable(rate_limited)
+            log_msg = f"Rate limit value stored in state: {rate_limit_string}. "
+            if rate_limited > now:
+                log_msg += "Still within rate limiting period, skipping task execution..."
+                self.logger.info(log_msg)
+                return [], state, False, 200, None
+
+            log_msg += "However no longer in rate limiting period, so task can be executed..."
+            del state[self.RATE_LIMIT_DATETIME]
+            self.logger.info(log_msg)
+
+    def check_rate_limit_error(self, status_code: int, state: dict) -> None:
+        if status_code == 429:
+            new_run_time = time() + 300
```

Comment:
On their docs it does look like they might return a header that contains when their rate limit is reset: https://developer.services.mimecast.com/api-overview - `X-Rate-Limit-Reset `. It might be better to base the time on this value if it exists?